core/src/main/java/org/jdbi/v3/core/ReduceRow.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core;
+
+import java.lang.reflect.Type;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.jdbi.v3.core.exception.UnableToExecuteStatementException;
+
+/**
+ * A ReduceRow is a view over a single row of a {@code ResultSet}
+ * that uses the current statement context to map rows and
+ * columns to values.  It is not valid past the return of
+ * the method that receives it.
+ */
+public class ReduceRow
```

Comment:
I don't think we can make this the default mapper.  It is actually not valid to "map" to a `ReduceRow` -- since it is a view, not a real object that holds its data.

I still think the current `Map<String, Object>` default sucks though.